src/main/kotlin/fr/xebia/xke/5_FP.kt

```diff
@@ -0,0 +1,73 @@
+package fr.xebia.xke
+
+import arrow.core.*
+
+import arrow.core.Either
+import arrow.core.flatMap
+import arrow.data.*
+
+/**
+ * This class validates user data in two ways:
+ * - Applicative for all fields (validated all fields)
+ * - Monadic within each field (stops on the first error per field)
+ *
+ *
+ * The following functions might help you
+ * - Option.fromNullable("something")
+ * - Option(maybeRight).toEither { "Left value" }
+ * - Try { data.toInt() }.toEither().mapLeft { NonEmptyList.of("$name must be an integer") }
```

Comment:
Even if the `Try` monad exist, I'll leave it as a possible alternative and not as part of the solution